packages/core/src/controllers/verify.ts

```diff
@@ -161,4 +139,83 @@ export class Verify extends IVerify {
     }
     return url;
   };
+
+  private fetchPublicKey = async () => {
+    this.logger.debug(`fetching public key from: ${this.verifyUrlV2}`);
+    const timeout = this.startAbortTimer(FIVE_SECONDS);
+    const result = await fetch(`${this.verifyUrlV2}/public-key`, {
+      signal: this.abortController.signal,
+    });
+    clearTimeout(timeout);
+    return (await result.json()) as jwk;
+  };
+
+  private persistPublicKey = async (publicKey: jwk) => {
+    this.logger.debug(`persisting public key to local storage`, publicKey);
+    await this.store.setItem(this.storeKey, publicKey);
+    this.publicKey = publicKey;
+  };
+
+  private removePublicKey = async () => {
+    this.logger.debug(`removing verify v2 public key from storage`);
+    await this.store.removeItem(this.storeKey);
+    this.publicKey = undefined;
+  };
+
+  private isValidJwtAttestation = async (attestation: string) => {
+    const key = await this.getPublicKey();
```

Comment:
Fetching the initial key when one doesn't exist yet should occur [at the same time as connecting to the relay with the intention of verifying a message](https://www.notion.so/walletconnect/Verify-Server-rewrite-tech-doc-Verify-API-V2-30e207c77f8447d8af37a186799b7c9e?pvs=4#64d9804559e74a988ab8540b5b508c6c), not after the message to verify is received. This avoids extra latency to lookup the key by doing it in-parallel.

Reply:
this was implemented according to your previous comment above

> 3\. Not run until there is an intention to use the public key. I.e. a pairing was just created on the wallet-side

will update

Reply:
My request is a bit nuanced:
- Not block SDK startup
- Not run on SDK startup
- Not block attestation validation
- Run concurrently with relay connection when a pairing is created

So I'd see it going like this:
- Pairing is requested
- Relay connection is established
- (concurrently) Public key is obtained if it currently is not fresh

Then when verification takes place, it should likely already have the public key. And no bandwidth is wasted on getting the public key when it's not going to be used by a pairing.